MIGRATION.md

```diff
@@ -18,6 +18,7 @@ API changes in MoveIt releases
 - `RobotTrajectory` provides a copy constructor that allows a shallow (default) and deep copy of waypoints
 - Replace the redundant namespaces `robot_state::` and `robot_model::` with the actual namespace `moveit::core::`. The additional namespaces will disappear in the future (ROS2).
 - Moved the library `moveit_cpp` to `moveit_ros_planning`. The classes are now in namespace `moveit_cpp`, access via `moveit::planning_interface` is deprecated.
+- The joint states of `passive` joints must be published in ROS and the CurrentStateMonitor will now wait for them as well. Their semantics dictate that they cannot be actively controlled, but they must be known to use the full robot state in collision checks. (https://github.com/ros-planning/moveit/pull/2663)
```

Comment:
@v4hn 
I have two problems with this:
1. I don't know what I actually need to change after reading this
2. I'm not sure it's actually true (and certainly not desirable) since you only iterate over `getActiveJointModels` below

(We had long discussions with IPA folks if a gripper should publish `gripper_opening` or rather `finger_left` (ie 0.5*opening) but I think nobody ever came up with having to publish `finger_right` (a mimic joint) as well)

Reply:
> I don't know what I actually need to change after reading this

Doesn't "The joint states of `passive` joints must be published in ROS" say what needs to be done?
Please propose a better wording for this entry if you think more is needed.

Are you sure this affects what you think it does? Mimic joints are **not** passive joints in MoveIt's terminology. Passive joints have to be declared in the srdf via `<passive_joint>`.
The latter are actually also part of the group of "active" joint models (yes I know this naming doesn't make much sense, I didn't pick it).
Passive joint are actively moving joints we can't update via mimic joints or can't even control on our own.
Levers or flexible joints that swing downwards towards gravity might be examples.

I'm happy to be contradicted and learn how you use this joint tag differently.

> We had long discussions with IPA folks if a gripper should publish gripper_opening or rather finger_left (ie 0.5*opening) 

Either is possible as long as the joint actuation interfaces accept the same joint(s).
All other relevant joints in the gripper should be computable via mimic joints.
If both fingers can be actuated independently that would of course be a different story.
It would be nice if we could actually have such coupled joint groups modelled in a way where specifying either is enough to infer the other (also for non-linear translations).

Reply:
Thanks! Now it makes sense ... somewhat ... I didn't know about the `passive` tag and category. I think people who use it will understand your migration notes. I misread this as mimic joints being required to be published and current with this PR